tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Grpc/GrpcLegacy/Client/GrpcLegacyClientCommon.cs

```diff
@@ -81,9 +81,9 @@ internal static class GrpcLegacyClientCommon
                 span.SetHeaderTags(requestMetadataWrapper, tracer.Settings.GrpcTagsInternal, GrpcCommon.RequestMetadataTagPrefix);
                 scope = tracer.ActivateSpan(span);
 
-                if (setSamplingPriority && existingSpanContext?.SamplingPriority is not null)
+                if (setSamplingPriority && existingSpanContext?.SamplingPriority is { } samplingPriority)
                 {
-                    span.Context.TraceContext?.SetSamplingPriority(existingSpanContext.SamplingPriority.Value);
+                    span.Context.TraceContext?.SetSamplingPriority(samplingPriority);
```

Comment:
probably a newbie question but just want to confirm that the `.Value` here is indeed not needed.

Reply:
Good question. The type of `SpanContext.SamplingPriority` is `int?`, so we needed `.Value` before to get an `int`. (Although, now that I look at `TraceContext.SetSamplingPriority()`, it does accept an `int?` so `.Value` wasn't really required.)

With the change to pattern matching syntax, the `samplingPriority` variable is of type `int`, so there is no `.Value` anymore.

Reply:
Ignoring `setSamplingPriority` for simplicity, these lines
```
if (existingSpanContext?.SamplingPriority is { } samplingPriority)
{
    <code>
```
translate roughly into
```
if (existingSpanContext != null && existingSpanContext.SamplingPriority != null)
{
    int samplingPriority = existingSpanContext.SamplingPriority.Value;
    <code>
```